### PR TITLE
[.NET] Update target frameworks

### PR DESCRIPTION
**Pull request by Romfos**
### 🤔 What's changed?

- Drop .NET 7 from test matrix
- Replace .NET 6 to .NET 8. because .NET 6 will reach End of Support on November 12, 2024
- Update C# language version to 12

### ⚡️ What's your motivation? 

- .NET 7 is out of support. https://devblogs.microsoft.com/dotnet/dotnet-7-end-of-support/
- .NET 6 (LTS) will reach End of Support on November 12, 2024 https://devblogs.microsoft.com/dotnet/dotnet-6-end-of-support/

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)
- :boom: Breaking change (incompatible changes to the API)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
- [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
- [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by Romfos** [comment]
please, squash PR during the merge

File: CHANGELOG.md
Comment (mpkorstanje) [inline CHANGELOG.md:13]: This should go under the `### Changed` header.

**Comment by mpkorstanje** [comment]
Cheers. I'm waiting for a review from either @gasparnagy or @clrudolphi.

Context: https://github.com/cucumber/messages/pull/237#issuecomment-2225810555

**Review by gasparnagy — APPROVED** [review]
Thx. I'm OK with this changes.

**Comment by gasparnagy** [comment]
@mpkorstanje these changes are "technically" breaking changes, but in practice they will not break any code, because we still have the support for ".NET Standard 2.0" which anyway covers all those older frameworks that are now removed from the "direct" support.

**Pull request merged**
